tests/cond_order/main.c

```diff
@@ -28,7 +28,7 @@
 #define THREAD_NUMOF            (5U)
 #define THREAD_FIRSTGROUP_NUMOF (3U)
 
-extern volatile thread_t *sched_active_thread;
+extern thread_t *sched_active_thread;
```

Comment:
I suggest we include sched.h instead of doing this declaration. (same for `tests/mutex_order`)

Reply:
I think there is an API call for that which should be used instead of accessing the variable directly

Reply:
If there is an API then quite some code is ignoring that API.

Reply:
I was wrong, there only is [`thread_get(kernel_pid_t pid)`](http://api.riot-os.org/group__core__thread.html#ga754515868f32530baa52ad2d05620205). One could use `thread_get(thread_getpid())`, but would be less efficient. Maybe anothrr static inline function should be added for that use case?

Reply:
> One could use `thread_get(thread_getpid())`, but would be less efficient. Maybe anothrr static inline function should be added for that use case?

+1 for another function. Maybe ```thread_get_current()```? That would be a nice PR of its own.

Reply:
Would it be an abuse to make  `thread_get(KERNEL_PID_UNDEF)` get the current thread?

Reply:
I'd say that `thread_get(KERNEL_PID_UNDEF)` is less readable than `thread_t * thread_get_current(void)`.